pinot-server/src/main/java/org/apache/pinot/server/starter/helix/HelixInstanceDataManager.java

```diff
@@ -194,7 +198,17 @@ private void reloadSegment(String tableNameWithType, SegmentMetadata segmentMeta
 
     File indexDir = segmentMetadata.getIndexDir();
     if (indexDir == null) {
-      LOGGER.info("Skip reloading REALTIME consuming segment: {} in table: {}", segmentName, tableNameWithType);
+      if (!_instanceDataManagerConfig.shouldReloadConsumingSegment()) {
+        LOGGER.info("Skip reloading REALTIME consuming segment: {} in table: {}", segmentName, tableNameWithType);
+        return;
+      }
+      LOGGER.info("Try reloading REALTIME consuming segment: {} in table: {}", segmentName, tableNameWithType);
+      SegmentMetadataImpl segmentMetadataImpl = (SegmentMetadataImpl) segmentMetadata;
```

Comment:
Just call the `reload` API on the segment, and let the segment decide if it is to be reloaded (i.e. depending on the config), and also how it should do it.

Reply:
Could you elaborate on what you mean? To me, this method is the implementation of `reload` API (i.e. "how it should do it"). The logics below indicate how immutable segments are reloaded. Here it's a bit different for consuming segments - we just compare the schema and update the newly added columns, without doing an actual segment reload/replace.

Reply:
I would bypass the references to DefaultColumnAction class altogether. Since we have a handle to the mutable index, just call the addExtraColumns API with the new schema. Just make sure that API handles well if we call it multiple times, either with same set of columns or newer ones than the previous time.

Agreed with your observation that this _is_ the reload API. It makes me think that reload API should be implemented as a part of the segment API, but that is a HUGE change, and I am not sure what else is involved there (and why it was done this way as a static method like ImmutableSegment.reload()).  So, let us not go into that.